Chatter/AvatarPicker/AvatarPickerRootView.swift

```diff
@@ -0,0 +1,135 @@
+//
+//  AvatarPickerRootView.swift
+//  Chatter
+//
+
+import Foundation
+
+public class AvatarPickerRootView: UIView {
+
+    typealias AvatarPickerDismissHandler = () -> Void
+    typealias AvatarTypeChangeHandler = (AvatarType) -> Void
+
+    var delegate: UICollectionViewDelegate? {
+        set {
+            collectionView.delegate = newValue
+        }
+        get {
+            return nil
+        }
+    }
+    var dataSource: UICollectionViewDataSource? {
+        set {
+            collectionView.dataSource = newValue
+        }
+        get {
+            return nil
+        }
+    }
+
+    var dismissHandler: AvatarPickerDismissHandler?
+    var avatarTypeChangeHandler: AvatarTypeChangeHandler?
+
+    private let backButton: UIButton = {
+        let backButton = UIButton(type: .custom)
+        backButton.setImage(#imageLiteral(resourceName: "smackBack"), for: .normal)
+        backButton.heightAnchor
+            .constraint(equalToConstant: 30)
+            .isActive = true
+        backButton.widthAnchor
+            .constraint(equalToConstant: 30)
+            .isActive = true
+        return backButton
+    }()
+
+    private let segmentedControl: UISegmentedControl = {
+        let segmentedControl = UISegmentedControl(items: ["Dark", "Light"])
+        segmentedControl.selectedSegmentIndex = 0
+        segmentedControl.heightAnchor
+            .constraint(equalToConstant: 30)
+            .isActive = true
+        segmentedControl.widthAnchor
+            .constraint(equalToConstant: 150)
+            .isActive = true
+        return segmentedControl
+    }()
+
+    private let collectionView: UICollectionView = {
+        let layout = UICollectionViewFlowLayout()
+        layout.scrollDirection = .vertical
+        let collectionView = UICollectionView(frame: .zero, collectionViewLayout: layout)
+        collectionView.backgroundColor = UIColor.white
+        collectionView.contentInset = UIEdgeInsets(top: 0, left: 20, bottom: 0, right: 20)
+        collectionView.register(AvatarPickerCell.self, forCellWithReuseIdentifier: "avatarCell")
```

Comment:
You can avoid setting explicit strings as the reuse identifier by using the name of the cell:
```
String(describing: AvatarPickerCell.self)
```